crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -587,6 +587,12 @@ impl CheckpointExecutor {
                     )
                     .await;
 
+                    #[cfg(msim)]
```

Comment:
@mystenmark I don't think we want to always do this in simtest. Is there a way to run this path probabilistically, maybe based on MSIM_TEST_SEED?

Reply:
You can use failpoints. See https://github.com/MystenLabs/sui/pull/10445/files for example.

Reply:
you can also just do `rand::random::<f64>() < 0.1` or whatever